### PR TITLE
lsp/refs: Just in time init for refs util

### PR DESCRIPTION
**Pull request by charlieegan3**
This was being slow to start and adding around 100ms to all regal command runs, even though this is only needed for the lsp.

This change makes the initialization of the pq var happen at the first call instead.

### CODE REVIEW

**Review by anderseknert — APPROVED** [review]
LGTM! I suppose pq isn't thread-safe, but worst case would be the initialize function is called more than once, which doesn't seem like a big deal.

**Review by srenatus — COMMENTED** [review]
I wouldn't have been surprised to see a `*sync.Once` being used here 🤔 But maybe we don't need this.

**Comment by anderseknert** [comment]
Since we both thought the same, let's use that :)

**Comment by charlieegan3** [comment]
Thanks both, have updated.

File: internal/lsp/completions/refs/used.go [hunk 1]
```diff
@@ -65,6 +73,10 @@ func init() {
 // See the rego above for more details on what's included and excluded.
 // This function is run when the parse completes for a module.
 func UsedInModule(ctx context.Context, module *ast.Module) ([]string, error) {
+	if pq == nil {
+		pqInitOnce.Do(initialize)
+	}
```
Comment (srenatus) [inline internal/lsp/completions/refs/used.go:78]: Should be save to do
```go
pqInitOnce.Do(initialize)
```
shouldn't it? 🤔

**Pull request merged**